akka-http2-support/src/main/scala/akka/http/impl/engine/http2/Http2Compliance.scala

```diff
@@ -4,17 +4,39 @@
 
 package akka.http.impl.engine.http2
 
+import scala.annotation.switch
+
 /** INTERNAL API */
 private[akka] object Http2Compliance {
+  def parseSettungsEnablePushValue(incomingValue: Int): Boolean =
+    (incomingValue: @switch) match {
+      case 0 ⇒ false
+      case 1 ⇒ true
+      case _ ⇒ throw new IllegalEnablePushValueSettingException(s"Illegal value in SETTINGS_ENABLE_PUSH settings header! MUST BE 0 or 1, but was: [$incomingValue]")
+    }
+
+  private val MinOutMaxFrameSize = Math.pow(2, 14).toInt // minimum allowed to be set
+  private val MaxOutMaxFrameSize = (Math.pow(2, 24) - 1).toInt // maximum allowed to be set
+
+  final class WindowOverflowException(currentWindow: Int, increment: Int)
```

Comment:
I wonder if we don't go a bit crazy with all those exception types. Do we really need them? What if we just have one or two exceptions, one for connection-level problems, and one for stream-level ones, that contain a message and an error code. Could still be created in here, WDYT?